Added a template for Streamlit Databricks data app (#56)
databricks/bundle-examples: contrib/templates/app-template/README.md

Reviewer: Instead of `app-template`, `streamlit-app`?

# app-template template

> This template contains a bundle structure to create and manage Databricks App which reads data from a SQL warehouse and visualizes it.

Reviewer: ```suggestion
This template contains a bundle structure for creating and managing a Databricks app that reads data from a SQL warehouse and visualizes it.
```

Looking at https://docs.databricks.com/en/dev-tools/databricks-apps/index.html, the capitalization rules appear to be "Databricks Apps" as the product name but a "Databricks app" when talking about a single instance.

The app is written in Python using Streamlit framework.

> Install the template using the following command:

Reviewer: ```suggestion
Initialize the template using the following command:
```
